server/container/spring/src/main/java/org/apache/james/container/spring/bean/factory/mailrepositorystore/MailRepositoryStoreBeanFactory.java

```diff
@@ -160,19 +160,12 @@ public synchronized MailRepository select(String destination) throws MailReposit
 
         String repID = destination;
         MailRepository reply = repositories.get(repID);
-        StringBuffer logBuffer;
         if (reply != null) {
-            if (LOGGER.isDebugEnabled()) {
-                logBuffer = new StringBuffer(128).append("obtained repository: ").append(repID).append(",").append(reply.getClass());
-                LOGGER.debug(logBuffer.toString());
-            }
+            LOGGER.debug("obtained repository: {},{}", repID, reply.getClass());
             return reply;
         } else {
             String repClass = classes.get(protocol);
-            if (LOGGER.isDebugEnabled()) {
-                logBuffer = new StringBuffer(128).append("obtained repository: ").append(repClass).append(" to handle: ").append(protocol).append(" with key ").append(protocol);
```

Comment:
I faithfully converted this to a parameterized logger call, but the log does not look right to me. Notice that `protocol` is inserted twice into the log message.

Reply:
Having looked at the code we can remove the "with key..." part IMO.

Reply:
Done